paimon-core/src/main/java/org/apache/paimon/mergetree/compact/PartialUpdateMergeFunction.java

```diff
@@ -145,64 +135,89 @@ private void updateNonNullFields(KeyValue kv) {
     private void updateWithSequenceGroup(KeyValue kv) {
         for (int i = 0; i < getters.length; i++) {
             Object field = getters[i].getFieldOrNull(kv.value());
-            SequenceGenerator sequenceGen = fieldSequences.get(i);
+            UserDefinedSeqComparator seqComparator = fieldSeqComparators.get(i);
             FieldAggregator aggregator = fieldAggregators.get(i);
             Object accumulator = getters[i].getFieldOrNull(row);
-            if (sequenceGen == null) {
+            if (seqComparator == null) {
                 if (aggregator != null) {
                     row.setField(i, aggregator.agg(accumulator, field));
                 } else if (field != null) {
                     row.setField(i, field);
                 }
             } else {
-                Long currentSeq = sequenceGen.generate(kv.value());
-                if (currentSeq != null) {
-                    Long previousSeq = sequenceGen.generate(row);
-                    if (previousSeq == null || currentSeq >= previousSeq) {
-                        row.setField(
-                                i, aggregator == null ? field : aggregator.agg(accumulator, field));
-                    } else if (aggregator != null) {
-                        row.setField(i, aggregator.agg(field, accumulator));
+                if (isEmptySequenceGroup(kv, seqComparator)) {
+                    // skip null sequence group
+                    continue;
+                }
+
+                if (seqComparator.compare(kv.value(), row) >= 0) {
+                    int index = i;
+
+                    // Multiple sequence fields should be updated at once.
+                    if (Arrays.stream(seqComparator.compareFields())
+                            .anyMatch(seqIndex -> seqIndex == index)) {
+                        for (int fieldIndex : seqComparator.compareFields()) {
+                            row.setField(
+                                    fieldIndex, getters[fieldIndex].getFieldOrNull(kv.value()));
+                        }
```

Comment:
The sequence fields will be updated multiple times when there are multiple compare fields